frontend/src/GameGrid/hooks/useGameGrid.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+
+type GridDataType = string[][]; // Define the type for your grid data
+
+const useFetchGrid = (url: string) => {
+    const [gridData, setGridData] = useState<GridDataType | null>(null);
+    const [loading, setLoading] = useState<boolean>(true);
+    const [error, setError] = useState<Error | null>(null);
+
+    useEffect(() => {
```

Comment:
If the component unmounts before the fetch request completes, it might try to set the state on an unmounted component, leading to a memory leak warning. To prevent this, you can use a cleanup function in your useEffect hook in the future:
```
.then(data => {
            if (isMounted) {
                setGridData(data); // Assuming response.json() returns the correct format
                setLoading(false);
            }
        })
        .catch(error => {
            if (isMounted) {
                setError(error);
                setLoading(false);
            }
        });
```